lib/bloc/weather_bloc.dart

```diff
@@ -14,19 +14,17 @@ class WeatherBloc extends Bloc<WeatherEvent, WeatherState> {
   WeatherBloc(this._weatherRepository) : super(WeatherInitial());
 
   @override
-  Stream<WeatherState> mapEventToState(
-      WeatherEvent event,
-      ) async* {
+  Stream<WeatherState> mapEventToState(WeatherEvent event) async* {
     if (event is GetWeather) {
       try {
         print("map events");
         yield WeatherLoading();
         final weather = await _weatherRepository.fetchWeather(event.cityName);
         print(weather.isRight());
-        yield weather.fold((l) => WeatherError("Couldn't fetch weather"), (r) =>  WeatherLoaded(r));
-      } on NetworkException {
+        yield weather.fold((l) => WeatherError("Couldn't fetch weather"), (r) => WeatherLoaded(r));
+      } catch (error, _) {
```

Comment:
I modified this to catch all, as it was any error other than a NetworkException resulted in a broken UI.